### ISSUE TITLE
[BUG] BaseDeepRegressorTorch mishandles (batch, 1) network output in loss and predict

### ISSUE THREAD

**Issue by srupat** [alan-turing-institute/sktime#10931]
**Describe the bug**

`BaseDeepRegressorTorch` yields targets of shape `(batch_size,)`, but four of the torch regressor networks end in `Linear(..., 1)` and emit `(batch_size, 1)`. This causes two separate bugs:

**1. Wrong training loss.** In `_run_epoch`, `self._criterion(y_pred, outputs)` broadcasts `(B, 1)` against `(B,)` into a `(B, B)` all-pairs matrix. The mean is taken over B² entries of which only the B diagonal ones are real residuals; gradients flow through the cross terms too, so this is a wrong training signal.

**2. Wrong predict shape.** `_predict` uses a bare `squeeze()`, which collapses `(1, 1)` to a 0-d array, so `predict()` on a single instance returns shape `()` instead of `(1,)`.

Affected: `FCNRegressorTorch`, `MLPRegressorTorch`, `SimpleRNNRegressorTorch`, `TapNetRegressorTorch`. The other six squeeze inside the network and are unaffected.

**To Reproduce**

```python
import numpy as np, torch
from sktime.regression.deep_learning import FCNRegressorTorch

X, y = np.random.rand(12, 2, 20), np.random.rand(12)
est = FCNRegressorTorch(num_epochs=1, batch_size=6, verbose=False, random_state=0)
est.fit(X, y)   # Raises UserWarning here

inputs, outputs = next(iter(est._build_dataloader(X, y)))
with torch.no_grad():
    yp = est.network(**inputs)
crit = torch.nn.MSELoss()
print(yp.shape, outputs.shape)                  # (6, 1) (6,)
print(crit(yp, outputs), crit(yp.squeeze(-1), outputs))   # broadcast vs correct

print(np.shape(est.predict(X[:1])))             # () -- expected (1,)
```

**Expected behavior**

The criterion should receive matching shapes, and `predict` should return shape `(n_instances,)` for any `n_instances`, including 1.

**Additional context**

The test suite cannot catch either one. `setup.cfg` sets `filterwarnings = ignore::UserWarning`, so the only signal the loss bug emits is discarded, and `_predict` squeezes on the way out so multi-instance output shape looks correct. `check_estimator` passes for all torch regressors either way.
